src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartBlockProducerAuRa.cs

```diff
@@ -207,31 +207,46 @@ bool CheckAddRandomnessTransactions(IList<ITxSource> list, IDictionary<long, Add
             return txSource;
         }
 
+        protected override ITxFilter CreateGasPriceTxFilter(ReadOnlyTxProcessorSource readOnlyTxProcessorSource)
+        {
+            ITxFilter gasPriceTxFilter = base.CreateGasPriceTxFilter(readOnlyTxProcessorSource);
+            Address? registrar = _context.ChainSpec?.Parameters.Registrar;
+            return registrar != null 
```

Comment:
can you possibly take the object creation outside of the parameters list - this is hard to read